wallet/views.py

```diff
@@ -1,23 +1,35 @@
 from django.shortcuts import render
 from django.http import HttpResponse
 from rest_framework import viewsets
-
+from django.db.models import Q
+from rest_framework.permissions import IsAuthenticated
 
 from wallet.models import Category, Expense
 from wallet.serializers import CategorySerializer, ExpenseSerializer
 
 
 class CategoryViewSet(viewsets.ModelViewSet):
-    queryset = Category.objects.all().order_by('-created_at')
     serializer_class = CategorySerializer
+    permission_classes = (IsAuthenticated,)
+
+    def get_queryset(self):
+        return Category.objects.filter(
+            Q(owner__isnull=True) |
+            Q(owner=self.request.user)
+        ).order_by('-id')
 
     def perform_create(self, serializer):
         serializer.save(owner=self.request.user)
 
 
 class ExpenseViewSet(viewsets.ModelViewSet):
-    queryset = Expense.objects.all().order_by('-created_at')
     serializer_class = ExpenseSerializer
+    permission_classes = (IsAuthenticated,)
+
+    def get_queryset(self):
+        return Expense.objects.filter(
+            owner=self.request.user
+        ).order_by('-id')
```

Comment:
are we sure about ordering by id?

Reply:
yes, what I want is ordering by created_at. And created_at is inserted automatically when a record was created, that means records were also inserted by ID.